relay-metrics/src/aggregation.rs

```diff
@@ -914,6 +919,7 @@ impl Message for FlushBuckets {
 /// }
 /// ```
 pub struct Aggregator {
+    public_key: ProjectKey,
```

Comment:
Having the public key here breaks our separation of concerns. However, the way the aggregator is structured right now needs to be revised anyway, so I'm happy to keep it in the interim.

@jjbayer we need to target this when proceeding with the metrics prototype. The simplest solution would be to parameterize the aggregator with data to pass through.